src/actionExecutor.ts

```diff
@@ -103,9 +104,35 @@ export class ActionExecutor {
     }
   }
 
+  private async handleRefund(
```

Comment:
🔧 Imho you are not _handling the refund_ but _wait for median blocktime_
```suggestion
  private async waitForBlockTime(
```

Reply:
refactored, please take a look again